src/test_helper.rs

```diff
@@ -0,0 +1,23 @@
+use std::path::PathBuf;
+use std::process::Command;
+
+use anyhow::{Context, Result};
+use clash::Clash;
+
+use super::*;
+
+pub fn sample_puzzle(name: &str) -> Result<Clash> {
+    let puzzle_file: PathBuf = ["fixtures", "puzzles", format!("{}.json", name).as_str()].iter().collect();
+    let contents = std::fs::read_to_string(&puzzle_file)
+        .with_context(|| format!("Unable to test puzzle with name {}", name))?;
+    let clash: Clash = serde_json::from_str(&contents)
+        .with_context(|| format!("Unable to deserialize test puzzle {} from {:?}", name, &puzzle_file))?;
+
+    Ok(clash)
+}
+
+pub fn sh_cmd(script: &str) -> Command {
+    let mut command = Command::new("sh");
+    command.arg("-c").arg(script);
+    command
+}
```

Comment:
I don't like having this thing at all, it feels dangerous even in tests. Couldn't we use `cat` and/or `echo` directly, without launching a shell? `Command::new("cat")` should pass the tests if the expected output is the same as input.

Reply:
Didn't think of `cat` to be honest. 

What would the plan be for (partially) failing test cases or solutions for more complex clashes?

I agree that it's dangerous. I'll see if I can make a safer alternative but I think it will include some stupid-looking code.